### DISCUSSION TITLE
How to use tmxlite on Windows and CLion

### DISCUSSION THREAD

**Discussion by jv-lime** [fallahn/tmxlite#142] [Q&A]
I'm new at using libraries and I didn't understand how to use the tmxlite. I tried to link the library in my IDE with many different ways, but none of them worked, and I didn't find any tutorial or issue that solved my problem.

**Comment by fallahn** [comment]
I've never used CLion but according to [this post](https://stackoverflow.com/questions/44998233/add-external-c-libraries-to-a-clion-project) it uses CMake files directly. In which case you can use the included example CMake files as a basis for adding tmxlite to your project:

https://github.com/fallahn/tmxlite/blob/master/SFMLExample/CMakeLists.txt#L25 for SFML projects or
https://github.com/fallahn/tmxlite/blob/master/OpenGLExample/CMakeLists.txt#L24 for SDL2 projects

**Comment by jv-lime** [comment]
I cloned the repository and tried to open the SFML Example in CLion, but it returned the following error:

```
CMake Error at C:/Program Files/JetBrains/CLion 2023.2/bin/cmake/win/x64/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find tmxlite (missing: TMXLITE_LIBRARIES)
Call Stack (most recent call first):
  C:/Program Files/JetBrains/CLion 2023.2/bin/cmake/win/x64/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  cmake/modules/FindTMXLITE.cmake:10 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:26 (find_package)
  ```

**Comment by fallahn** [comment]
You need to build the tmxlite library first and install it somewhere that CMake can find it. You can do this with the CMake file in the tmxlite directory.
